packages/suite/src/actions/wallet/send/sendFormThunks.ts

```diff
@@ -80,44 +90,142 @@ export const importSendFormRequestThunk = createThunk(
     (_, { dispatch }) => dispatch(modalActions.openDeferredModal({ type: 'import-transaction' })),
 );
 
+const updateRbfLabelsThunk = createThunk(
+    `${MODULE_PREFIX}/updateReplacedTransactionThunk`,
+    (
+        {
+            labelsToBeEdited,
+            precomposedTx,
+            txid,
+        }: {
+            labelsToBeEdited: RbfLabelsToBeUpdated;
+            precomposedTx: PrecomposedTransactionFinal;
+            txid: string;
+        },
+        { dispatch, extra },
+    ) => {
+        const {
+            thunks: { moveLabelsForRbfAction },
+        } = extra;
+
+        dispatch(
+            moveLabelsForRbfAction({
+                toBeMovedOrDeletedList: labelsToBeEdited,
+                newTxid: txid,
+            }),
+        );
+
+        // notification from the backend may be delayed.
+        // modify affected transaction(s) in the reducer until the real account update occurs.
+        // this will update transaction details (like time, fee etc.)
+        dispatch(
+            replaceTransactionThunk({
+                precomposedTx,
+                newTxid: txid,
+            }),
+        );
+    },
+);
+
+const applySendFormMetadataLabelsThunk = createThunk(
+    `${MODULE_PREFIX}/applyMetadataLabelsThunk`,
+    (
+        {
+            selectedAccount,
+            precomposedTx,
+            txid,
+        }: {
+            selectedAccount: Account;
+            precomposedTx: GeneralPrecomposedTransactionFinal;
+            txid: string;
+        },
+        { dispatch, getState },
+    ) => {
+        const metadata = selectMetadata(getState());
+        if (metadata.enabled) {
+            const formDraft = selectSendFormDraftByAccountKey(getState(), selectedAccount.key);
+            let outputsPermutation: number[];
+            if (isCardanoTx(selectedAccount, precomposedTx)) {
+                // cardano preserves order of outputs
+                outputsPermutation = precomposedTx?.outputs.map((_o, i) => i);
+            } else {
+                outputsPermutation = precomposedTx?.outputsPermutation;
+            }
```

Comment:
I got your concer about the unneccessary `let`. Since it is only three lines of code and the new function would not be used anywhere else, I decided to change it to ternary operator call instead, Fixed by 7b2265b66b.